src/virtualenv/util/lock.py

```diff
@@ -14,7 +14,7 @@
 class _CountedFileLock(FileLock):
     def __init__(self, lock_file):
         parent = os.path.dirname(lock_file)
-        if not os.path.exists(parent):
+        if not os.path.isdir(parent):
```

Comment:
should only check that it's a directory, not in general

there's also a TOCTOU bug here but it's unlikely to be hit